TabbedSplitViewController/TabbedSplitViewController/TabbedSplitMainView.swift

```diff
@@ -103,34 +107,164 @@ class PKTabbedSplitView: UIView {
         fatalError("init(coder:) has not been implemented")
     }
 
+    // MARK: - Presenting detail view in-place
+
+    /// Present detail view in-place, hiding master and detail, if not already hidden
+    func presentDetailViewSolo(hidingTabBar: Bool, hidingMaster: Bool) {
+        let masterNewFrame = hidingMaster ? newFrameWhenHiding(for: .master) : nil
+        let tabBarNewFrame = hidingTabBar ? newFrameWhenHiding(for: .tabBar) : nil
+
+        let view = self.view(for: .detail)
+        stackView.insertSubview(view, at: StackViewItem.detail.hierarchyIndex)
+
+        // Set frame to full screen and hide behind right edge
+        view.frame = stackView.frame
+        view.frame.origin.x = stackView.frame.maxX
+        view.isHidden = false
+
+        UIView.animate(withDuration: 0.33, animations: {
+            if let newFrame = masterNewFrame {
+                self.view(for: .master).frame = newFrame
+            }
+            if let newFrame = tabBarNewFrame {
+                self.view(for: .tabBar).frame = newFrame
+            }
+            view.frame.origin.x = 0
+        }) { _ in
+            self.view(for: .master).isHidden = true
+            self.view(for: .tabBar).isHidden = true
+            self.addArrangedView(.detail)
+        }
+    }
+
+    /// Hide detail view in-place, showing tab bar and master
+    ///  if they were hidden by `presentDetailViewSolo(hidingTabBar:hidingMaster:)`
+    ///  but otherwise should be shown
+    ///
+    /// - Parameters:
+    ///   - keepShown: keep detail view on screen
+    ///   - addingTabBar: add the tab bar back
+    ///   - addingMaster: add the master view back
+    func hideDetailViewSolo(keepShown: Bool, addingTabBar: Bool, addingMaster: Bool) {
+        func prepareView(for item: StackViewItem) {
+            let view = self.view(for: item)
+            stackView.insertSubview(view, at: item.hierarchyIndex)
+            view.frame.size.height = stackView.frame.height
+            view.frame.origin.x = -view.frame.size.width
```

Comment:
Detail view should initially appear on the right side (see `newFrameWhenHiding`)
```suggestion
            view.frame.origin.x = stackView.frame.size.width
```